lockfile/src/golang.rs

```diff
@@ -37,6 +37,50 @@ impl Parse for GoSum {
     }
 }
 
+pub struct GoDeps {
+    pub go: String,
+    pub modules: Vec<Package>,
+}
+
+fn check_go_directive(version: &str) -> anyhow::Result<()> {
+    let numeric_part = version.split(|c: char| !c.is_numeric()).collect::<Vec<&str>>();
+    if let (Some(major), Some(minor)) = (numeric_part.first(), numeric_part.get(1)) {
+        let major: u32 = major.parse().unwrap_or(0);
+        let minor: u32 = minor.parse().unwrap_or(0);
+
+        // Check if version meets the criteria.
+        if major < 1 || minor < 17 {
+            bail!("Minimum supported go directive is 1.17")
+        }
+    } else {
+        bail!("Error parsing go directive")
```

Comment:
There's currently only 4 instances of `bail!` in CLI, so it is not frequently used. None of these were written by me either.

What is the advantage of using a non-obvious macro here over standard Rust functionality? How is a new contributor supposed to distinguish between `bail!` and `panic!` if all the relevant bits of what `bail!` does are hidden inside the macro?

Reply:
Yeah it's just harder to read imo.